### PR TITLE
Add microwave oven control cluster xml

### PR DESCRIPTION
**Pull request by fengdx4**
Problem:
Matter 1.3 clusters missing XML and generated code for Microwave Oven Control cluster

Solution:
Create XML
Update below files:
.github/workflows/tests.yaml
scripts/rules.matterlint
src/app/zap-templates/zcl/data-model/all.xml
src/app/zap-templates/zcl/zcl-with-test-extensions.json
src/app/zap-templates/zcl/zcl.json
src/app/zap_cluster_list.json
src/controller/data_model/BUILD.gn
src/controller/data_model/controller-clusters.zap

Add new cluster XML
src/app/zap-templates/zcl/data-model/chip/microwave-oven-mode-cluster.xml

Testing:
Regen all
Rebuild chip-tool

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <cluster>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml:20]: Should this have `<cluster apiMaturity="provisional">` until this cluster is finalized (i.e. passes SVE)?

File: examples/air-quality-sensor-app/air-quality-sensor-common/air-quality-sensor-app.zap [hunk 2]
```diff
@@ -3243,7 +3243,7 @@
               "code": 2,
               "mfgCode": null,
               "side": "server",
-              "type": "int16u",
+              "type": "vendor_id",
```
Comment (bzbarsky-apple) [inline examples/air-quality-sensor-app/air-quality-sensor-common/air-quality-sensor-app.zap:3246]: These look like changes unrelated to this PR. If this is from running zap_convert_all.py, could you please do a separate PR for that?

File: src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml:20]: Yes.

File: src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <name>Microwave Oven Control</name>
+    <domain>Appliances</domain>
+    <description>Attributes and commands for configuring the microwave oven control, and reporting cooking stats.</description>
+    <code>0x050F</code>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml:24]: This happens to be the wrong cluster ID, though it's hard to see that from the spec for the cluster, where it's also wrong.  See https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7779

File: src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    <attribute side="server" code="0x0002" define="POWER_SETTING" type="int8u" writable="false" optional="false">PowerSetting</attribute>
+    <attribute side="server" code="0x0003" define="MIN_POWER" type="int8u" writable="false" optional="true">MinPower</attribute>
+    <attribute side="server" code="0x0004" define="MAX_POWER" type="int8u" writable="false" optional="true">MaxPower</attribute>
+    <attribute side="server" code="0x0005" define="POWERSTEP" type="int8u" writable="false" optional="true">PowerStep</attribute>
+
+    <command source="client" code="0x00" name="SetCookingParameters" optional="false">
+        <description>Set Cooking Parameters</description>
+        <arg name="CookMode" type="int8u" optional="true"/>
+        <arg name="CookTime" type="elapsed_s" optional="true"/>
+        <arg name="PowerSetting" type="int8u" min="MIN_POWER" max="MAX_POWER" optional="true"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/microwave-oven-control-cluster.xml:38]: Pretty sure those min/max values would not be understood if anything were trying to use them, but nothing is right now.

File: src/app/zap-templates/zcl/zcl-with-test-extensions.json [hunk 5]
```diff
@@ -323,6 +324,14 @@
             "OnMode",
             "FeatureMap"
         ],
+        "Microwave Oven Control": [
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl-with-test-extensions.json:327]: This is part of the implementation, not the cluster XML.  Until we have implementation, we should not be adding anything describing that implementation here.

File: src/app/zap-templates/zcl/zcl.json [hunk 6]
```diff
@@ -314,6 +315,14 @@
             "DSTOffsetListMaxSize"
         ],
         "Temperature Control": ["SupportedTemperatureLevels"],
+        "Microwave Oven Control": [
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.json:318]: Again, this should not be here.

File: src/app/zap_cluster_list.json [hunk 7]
```diff
@@ -182,6 +183,7 @@
         "LOW_POWER_CLUSTER": ["low-power-server"],
         "MEDIA_INPUT_CLUSTER": ["media-input-server"],
         "MEDIA_PLAYBACK_CLUSTER": ["media-playback-server"],
+        "MICROWAVE_OVEN_CONTROL_CLUSTER": ["microwave-oven-control-server"],
```
Comment (bzbarsky-apple) [inline src/app/zap_cluster_list.json:186]: Not sure we need these pieces until we are implementing either...  But if we do, ok.

**Review by bzbarsky-apple — APPROVED** [review]
